contracts/test-utils/src/assertions.ts

```diff
@@ -1,16 +1,50 @@
+import { abisToAbiBySelector, decodeCallData } from '@0x/order-utils';
 import { RevertReason } from '@0x/types';
-import { logUtils } from '@0x/utils';
+import { AbiEncoder, logUtils } from '@0x/utils';
 import { NodeType } from '@0x/web3-wrapper';
 import * as chai from 'chai';
-import { TransactionReceipt, TransactionReceiptStatus, TransactionReceiptWithDecodedLogs } from 'ethereum-types';
+import {
+    MethodAbi,
+    TransactionReceipt,
+    TransactionReceiptStatus,
+    TransactionReceiptWithDecodedLogs,
+} from 'ethereum-types';
+
+import * as ethers from 'ethers';
 import * as _ from 'lodash';
 
+import { constants } from './constants';
 import { web3Wrapper } from './web3_wrapper';
 
 const expect = chai.expect;
 
 let nodeType: NodeType | undefined;
 
+const ERROR_ABIS = _.map(
+    constants.ERROR_ABI_STRINGS,
+    (errorAbiString: string) => ethers.utils.parseSignature(errorAbiString) as MethodAbi,
+);
+const errorAbiBySelector = abisToAbiBySelector(ERROR_ABIS);
+const solidityRevertSelector = new AbiEncoder.Method(ethers.utils.parseSignature(
+    constants.SOLIDITY_REVERT_ABI,
+) as MethodAbi).getSelector();
+const bytesRevertAbi = ethers.utils.parseSignature(constants.BYTES_REVERT_ABI) as MethodAbi;
+/**
+ * HACK: By default solidity encodes revert reasons as `Error(string)` we want to throw additional data
+ * so we encode that data using an error signature e.g. `function INVALID_ORDER_SIGNATURE(bytes32 orderHash)`.
+ * After that we treat those bytes as string and use it instead of a revert reason. When we try to decode that malformed revert reason - we have a selector - ABI mismatch.
```

Comment:
```suggestion
 * We pretend those encoded bytes are a string (Solidity lets us do that) and use it instead of a string revert reason. 
 * In order for us to decode the revert properly, we need to use ABI `Error(bytes)` NOT `Error(string)`. Solidity by default
 * uses the selector for `Error(string)`, so we introduce a hack where we map that selector to the ABI definition for `Error(bytes)`. 
 * Doing so let's us decode the revert string AS BYTES! Which is what we need to do to avoid a selector <-> ABI mismatch.
 
```